src/components/views/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,40 @@
+import {Text, View} from 'react-native';
+import Logo from '../../../static/imgs/rst_logo.svg';
+import {styles} from './style';
+import {useState} from 'react';
+import {BackArrow, Login, Main, Register} from '../../../components';
+import auth from '@react-native-firebase/auth';
+
+export const MainPage = () => {
+  const [view, setView] = useState<'login' | 'main' | 'register'>('main');
+  const goBack = () => {
+    setView('main');
+    auth()
+      .signOut()
+      .then(() => alert('wylogowano'));
+  };
+  const renderView = () => {
```

Comment:
Co tu się dzieje tej kurwa. React-navigation dodaj kurwa, a nie jakaś czarną magię robisz. Muszą być widoki normalnie, a nie jakaś podmiana komponentów tej

Reply:
ok ale tutaj chujowo wygląda przechodzenie, jakbym chciał zanimować tylko te guziki na dole i formularze przy pojawianu się to nie ogarnę tego react-navigation